superduperdb/base/datalayer.py

```diff
@@ -908,6 +907,29 @@ def _add(
         jobs.extend(these_jobs)
         return jobs
 
+    def _change_component_reference_prefix(self, serialized):
+        """Replace '?' to '&' in the serialized object."""
+        references = set()
+        for reference in list(serialized['_leaves'].keys()):
+            if isinstance(serialized['_leaves'][reference], Component):
+                serialized['_leaves'].pop(reference)
+                references.add(reference)
+
+        # Only replace component references
+        if not references:
+            return
+
+        def replace_function(value):
```

Comment:
It is actually the value, because as a key in _leaves it has been deleted, it only matches the value under the pattern

`{'child': '?:component:/xxxxx'}`, which is transformed into `{'child': '&:component:/xxxxx'}`.